File: 1508-range-sum-of-sorted-subarray-sums/1508-range-sum-of-sorted-subarray-sums.java
class Solution {
    public int rangeSum(int[] nums, int n, int left, int right) {
        
        int mod = (int)1e9+7;
            
        int[] arr= new int[n*(n+1)/2];
        int k= 0;
        
        for (int i= 0; i< n; i++) {
            int acc= 0;
            for (int j=i; j<n; j++) {
                acc += nums[j];
                arr[k] = acc;
                k++;
            }
        }
        
        Arrays.sort(arr);
        int res= 0;
        
        for (int i= left-1; i< right; i++) {
            res = (res+ arr[i])% mod;
        }
        
        return res;
        
    }
}

File: 1508-range-sum-of-sorted-subarray-sums/1508-range-sum-of-sorted-subarray-sums.py
class Solution:
    def rangeSum(self, nums: List[int], n: int, left: int, right: int) -> int:
        l1 = []
        for i in range(n):
            total = nums[i]
            for j in range(i,n-1):
                l1.append(total)
                total += nums[j+1]
            l1.append(total)
        l1.sort()
        output = 0
        while left <= right:
            output += l1[left-1]
            left += 1
        return output%1000000007